plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -341,6 +345,29 @@ public String[] getLocalDeployedModels() {
             .toArray(new String[0]);
     }
 
+    /**
+     * Get expired models on node.
+     *
+     * @return array of expired model id
+     */
+    public String[] getExpiredModels() {
+        return modelCaches.entrySet().stream().filter(entry -> {
+            MLModel mlModel = entry.getValue().getCachedModelInfo();
```

Comment:
Cached model info only set when predict request comes. Check https://github.com/opensearch-project/ml-commons/pull/1472/files

For a deployed model, if no predict request comes, the cached model info is null.  Then the `mlModel` will be null and line 356 will throw null pointer exception, then sync up cron job will get no response.

Reply:
Add this line `modelCacheHelper.setModelInfo(modelId, mlModel);` when deploy model should fix the issue